### PR TITLE
Update sharedTree fuzz tests to use simple-tree, and remove flex-tree usages

### PR DESCRIPTION
**Pull request by daesunp**
## Description

This PR updates the sharedTree fuzz tests to use the simple tree apis, and removes the use of flex-tree. The schema op is currently disabled, as we manually dispose of the view in these tests during schema updates, and run into a known double dispose issue.

### CODE REVIEW

File: packages/dds/tree/src/test/shared-tree/fuzz/anchorStability.fuzz.spec.ts
Comment (taylorsw04) [inline packages/dds/tree/src/test/shared-tree/fuzz/anchorStability.fuzz.spec.ts:67]: you should link https://dev.azure.com/fluidframework/internal/_workitems/edit/11436

File: packages/dds/tree/src/test/shared-tree/fuzz/fuzzUtils.ts
Comment (taylorsw04) [inline packages/dds/tree/src/test/shared-tree/fuzz/fuzzUtils.ts:71]: Should this be "ArrayChildren" or something to avoid a reference to an implementation detail (sequences)?

File: packages/dds/tree/src/test/shared-tree/fuzz/fuzzUtils.ts
Comment (taylorsw04) [inline packages/dds/tree/src/test/shared-tree/fuzz/fuzzUtils.ts:104]: remove sequence naming here.

File: packages/dds/tree/src/test/shared-tree/fuzz/fuzzUtils.ts
Comment (taylorsw04) [inline packages/dds/tree/src/test/shared-tree/fuzz/fuzzUtils.ts:100]: what is this function for? why is there a second schema? can we doc it?

File: packages/dds/tree/src/test/shared-tree/fuzz/anchorStability.fuzz.spec.ts
Comment (taylorsw04) [inline packages/dds/tree/src/test/shared-tree/fuzz/anchorStability.fuzz.spec.ts:47]: why aren't we able to type this as FuzzNode and use the FuzzNode constructor (passing in the same json)?

File: packages/dds/tree/src/test/shared-tree/fuzz/composeVsIndividual.fuzz.spec.ts
Comment (taylorsw04) [inline packages/dds/tree/src/test/shared-tree/fuzz/composeVsIndividual.fuzz.spec.ts:112]: commenting the body of this method (it isn't clear why you're disposing the local branch and using a fork, or why we are assigning a node schema to the view) would be helpful

File: packages/dds/tree/src/test/shared-tree/fuzz/composeVsIndividual.fuzz.spec.ts
Comment (taylorsw04) [inline packages/dds/tree/src/test/shared-tree/fuzz/composeVsIndividual.fuzz.spec.ts:182]: why don't we dispose the branch, like in the similar code above?

File: packages/dds/tree/src/test/shared-tree/fuzz/composeVsIndividual.fuzz.spec.ts
Comment (taylorsw04) [inline packages/dds/tree/src/test/shared-tree/fuzz/composeVsIndividual.fuzz.spec.ts:195]: could all this setup share code with above?

File: packages/dds/tree/src/test/shared-tree/fuzz/composeVsIndividual.fuzz.spec.ts
Comment (taylorsw04) [inline packages/dds/tree/src/test/shared-tree/fuzz/composeVsIndividual.fuzz.spec.ts:196]: we leave a transaction open on test start?

File: packages/dds/tree/src/test/shared-tree/fuzz/fuzzEditReducers.ts
Comment (taylorsw04) [inline packages/dds/tree/src/test/shared-tree/fuzz/fuzzEditReducers.ts:155]: do we not need to put the new view in the map? (I'm not sure how this map is used, feel free to ignore)

File: packages/dds/tree/src/test/shared-tree/fuzz/fuzzEditReducers.ts
Comment (taylorsw04) [inline packages/dds/tree/src/test/shared-tree/fuzz/fuzzEditReducers.ts:218]: remove?

File: packages/dds/tree/src/test/shared-tree/fuzz/fuzzEditReducers.ts
Comment (taylorsw04) [inline packages/dds/tree/src/test/shared-tree/fuzz/fuzzEditReducers.ts:410]: this is unused I believe

File: packages/dds/tree/src/test/shared-tree/fuzz/anchorStability.fuzz.spec.ts
Comment (daesunp) [reply packages/dds/tree/src/test/shared-tree/fuzz/anchorStability.fuzz.spec.ts]: Updated work item, and added comments to fuzz tests linking to the work item

File: packages/dds/tree/src/test/shared-tree/fuzz/fuzzUtils.ts
Comment (daesunp) [reply packages/dds/tree/src/test/shared-tree/fuzz/fuzzUtils.ts]: Updated `SequenceChildren` to `ArrayChildren`

File: packages/dds/tree/src/test/shared-tree/fuzz/fuzzUtils.ts
Comment (daesunp) [reply packages/dds/tree/src/test/shared-tree/fuzz/fuzzUtils.ts]: removed sequence naming

File: packages/dds/tree/src/test/shared-tree/fuzz/fuzzUtils.ts
Comment (daesunp) [reply packages/dds/tree/src/test/shared-tree/fuzz/fuzzUtils.ts]: This function is only called in `createTreeViewSchema`, so it shouldn't be called multiple times with the same schemaFactory. `createTreeViewSchema` makes its own instance of `SchemaFactory`, so it should be okay. I have removed the `export` from `createFuzzNodeSchema`, and added doc comments for both `createFuzzNodeSchema` and `createTreeViewSchema`

File: packages/dds/tree/src/test/shared-tree/fuzz/anchorStability.fuzz.spec.ts
Comment (daesunp) [reply packages/dds/tree/src/test/shared-tree/fuzz/anchorStability.fuzz.spec.ts:47]: Unfortunately since we are using `FuzzNode` for casting during schema comparisons, and creating different schema (which satisfies the structure of `FuzzNode`) in the fuzzViews (as during updates it creates a new schema), creating an instance of the `FuzzNode` leads to a `document out of schema` error

File: packages/dds/tree/src/test/shared-tree/fuzz/composeVsIndividual.fuzz.spec.ts
Comment (daesunp) [reply packages/dds/tree/src/test/shared-tree/fuzz/composeVsIndividual.fuzz.spec.ts]: After chatting with @CraigMacomber and @yann-achard-MS we decided to remove schemaOps from this test, as there was no good way of continuing the transaction after applying a schemaOp (since we create a new view from the fork when applying the schemaOp).

File: packages/dds/tree/src/test/shared-tree/fuzz/composeVsIndividual.fuzz.spec.ts
Comment (daesunp) [reply packages/dds/tree/src/test/shared-tree/fuzz/composeVsIndividual.fuzz.spec.ts:153]: For these tests we keep track of two views (one for main, and one for the fork), where all of the edits are applied to the forked view. We disposed of the forked view when applying schemaOps, since the new forked view will have the updated schema.

But now that logic has been deleted, since we removed schemaOps from these tests.

File: packages/dds/tree/src/test/shared-tree/fuzz/composeVsIndividual.fuzz.spec.ts
Comment (daesunp) [reply packages/dds/tree/src/test/shared-tree/fuzz/composeVsIndividual.fuzz.spec.ts:166]: Removed logic from code above, as we decided to remove schemaOps from the composeVsIndividual tests

File: packages/dds/tree/src/test/shared-tree/fuzz/composeVsIndividual.fuzz.spec.ts
Comment (daesunp) [reply packages/dds/tree/src/test/shared-tree/fuzz/composeVsIndividual.fuzz.spec.ts:167]: The `composeVsIndividual` tests start the transaction on the fork at the beginning of the test, and commit them at the end of the test (with the transaction weights set to 0), so that it composes the edits. After committing the transaction, we merge the fork into the main, and check that we end up with the same state.

File: packages/dds/tree/src/test/shared-tree/fuzz/fuzzEditReducers.ts
Comment (daesunp) [reply packages/dds/tree/src/test/shared-tree/fuzz/fuzzEditReducers.ts:178]: Updated to put the new view in the map

File: packages/dds/tree/src/test/shared-tree/fuzz/fuzzEditReducers.ts
Comment (daesunp) [reply packages/dds/tree/src/test/shared-tree/fuzz/fuzzEditReducers.ts]: removed unused function

File: packages/dds/tree/src/test/shared-tree/fuzz/fuzzEditReducers.ts
Comment (daesunp) [reply packages/dds/tree/src/test/shared-tree/fuzz/fuzzEditReducers.ts]: removed unnecessary call to `Tree.parent()`

File: packages/dds/tree/src/test/shared-tree/fuzz/anchorStability.fuzz.spec.ts
Comment (CraigMacomber) [reply packages/dds/tree/src/test/shared-tree/fuzz/anchorStability.fuzz.spec.ts:47]: This us super gross and unsafe, but I think justified. That said, I think it should get a comment explaining why it can't use unhydrated nodes, and also why is can't be type safe (needs that type cast at the end), since neither of those limitation are obvious and in almost all cases either of those would be bad.

File: packages/dds/tree/src/test/shared-tree/fuzz/fuzzEditReducers.ts
Comment (yann-achard-MS) [inline packages/dds/tree/src/test/shared-tree/fuzz/fuzzEditReducers.ts:247]: This is inserting the nodes one by one, which is not a interesting for testing purposes as inserting them in one block. You may be able to use `insertAt(index, TreeArrayNode.spread(insertValues))` to do that.

File: packages/dds/tree/src/test/shared-tree/fuzz/fuzzEditReducers.ts
Comment (yann-achard-MS) [inline packages/dds/tree/src/test/shared-tree/fuzz/fuzzEditReducers.ts:452]: I'm finding this name misleading since it sometimes generate a non-leaf node. How about just `generateFuzzNode`?

File: packages/dds/tree/src/test/shared-tree/fuzz/operationTypes.ts
Comment (yann-achard-MS) [inline packages/dds/tree/src/test/shared-tree/fuzz/operationTypes.ts:149]:
```suggestion
	dstParent: DownPath | undefined;
```
Since this isn't describing the path to a field anymore.

File: packages/dds/tree/src/test/shared-tree/fuzz/topLevel.fuzz.spec.ts
Comment (yann-achard-MS) [inline packages/dds/tree/src/test/shared-tree/fuzz/topLevel.fuzz.spec.ts:101]: I'm guessing this was left in here by mistake?

File: packages/dds/tree/src/test/shared-tree/fuzz/fuzzUtils.ts
Comment (yann-achard-MS) [inline packages/dds/tree/src/test/shared-tree/fuzz/fuzzUtils.ts:54]: What's the motivation for having these instead of directly using string, number, and handle? I.e., why wrap them like this?

File: packages/dds/tree/src/test/shared-tree/fuzz/fuzzEditReducers.ts
Comment (taylorsw04) [reply packages/dds/tree/src/test/shared-tree/fuzz/fuzzEditReducers.ts:178]: I'm surprised nothing broke when this was the case. Why is that?

Comment (daesunp) [reply packages/dds/tree/src/test/shared-tree/fuzz/fuzzEditReducers.ts:178]: It was because the schemaOps were disabled for the tests (as there is currently the double dispose error we run into), and when enabled, the test fails as soon as we call `upgradeSchema`. However, I forgot that the `viewFromState` function checks the map for a cached view, and otherwise creates a new view to return anyways. I am just going to double check this to see if we actually need to store this newView in the map

File: packages/dds/tree/src/test/shared-tree/fuzz/fuzzEditReducers.ts
Comment (daesunp) [reply packages/dds/tree/src/test/shared-tree/fuzz/fuzzEditReducers.ts]: Updated to insert values in one block

File: packages/dds/tree/src/test/shared-tree/fuzz/fuzzEditReducers.ts
Comment (daesunp) [reply packages/dds/tree/src/test/shared-tree/fuzz/fuzzEditReducers.ts]: Updated name to `generateFuzzNode`

File: packages/dds/tree/src/test/shared-tree/fuzz/topLevel.fuzz.spec.ts
Comment (daesunp) [reply packages/dds/tree/src/test/shared-tree/fuzz/topLevel.fuzz.spec.ts]: Whoops, it was left in there accidentally during debugging. Removed `skipMinimization: true`

File: packages/dds/tree/src/test/shared-tree/fuzz/fuzzUtils.ts
Comment (daesunp) [reply packages/dds/tree/src/test/shared-tree/fuzz/fuzzUtils.ts:54]: The motivation behind this was originally to be able to call `Tree.parent` to get the field path, but decided to just use grab the `UpPath` from the anchorNode and convert that to the `DownPath` instead. 

However currently we are checking that a selected node is of type `FuzzNode` when selecting a node for the fieldEdit, and wrapping them in a node allowed made calling `Tree.is()` on the node easier, so I left it as is

File: packages/dds/tree/src/test/shared-tree/fuzz/fuzzEditReducers.ts
Comment (daesunp) [reply packages/dds/tree/src/test/shared-tree/fuzz/fuzzEditReducers.ts:179]: I wasn't able to check in the fuzz tests directly (as it fails as soon as we call `upgradeSchema` for known view disposal issues), but re-created a similar scenario in a test and checked that the tree's storedSchema does get updated once we call `upgradeSchema` on its view.

Since the `viewFromState` function will recreate a new view (getting a simple tree schema from the tree's storedSchema), it should work correctly even if we don't store this newView in the map. That being said, it would also be fine if we set the view in the map as well.

Did you want me to remove this bit of code?

Comment (taylorsw04) [reply packages/dds/tree/src/test/shared-tree/fuzz/fuzzEditReducers.ts:179]: I'll leave it up to you :)

Comment (daesunp) [reply packages/dds/tree/src/test/shared-tree/fuzz/fuzzEditReducers.ts:179]: Sounds good Taylor :)

File: packages/dds/tree/src/test/shared-tree/fuzz/fuzzUtils.ts
Comment (yann-achard-MS) [reply packages/dds/tree/src/test/shared-tree/fuzz/fuzzUtils.ts:54]: Also, if there's something awkward about `Tree.is()` and primitives then we probably need to fix that instead of getting around the issue.

Comment (daesunp) [reply packages/dds/tree/src/test/shared-tree/fuzz/fuzzUtils.ts]: Sorry Yann, just checked that `Tree.is()` is working properly with primitives, and I was mistaken :) 

Updated the fuzz leaf types to be primitives

**Review by taylorsw04 — APPROVED** [review]

**Pull request merged**
